scvi/module/_mrdeconv.py

```diff
@@ -244,17 +267,24 @@ def loss(
             var_vprior = torch.transpose(self.var_vprior, 0, 1).unsqueeze(
                 0
             )  # 1, p, n_labels, n_latent
-            pre_lse = (
-                Normal(mean_vprior, torch.sqrt(var_vprior)).log_prob(gamma).sum(-1)
+            weight_vprior = torch.transpose(self.weight_vprior, 0, 1)  # p, n_labels
+            pre_lse = weight_vprior * torch.exp(
+                Normal(mean_vprior, torch.sqrt(var_vprior) + 1e-4)
+                .log_prob(gamma)
+                .sum(-1)
```

Comment:
@romain-lopez It's LogSumExp in the original code. However, this is my custom function of weighted LogSumExp (not supported by PyTorch). I get the idea of sum instead of multiply. However, it would often lead to adding log(0) in cases where weight_vprior is 0, right? What's your suggestion to put there? Overall, you are right, that the current version is numerically less stable than LogSumExp and I am open to adapt it.

Reply:
maybe you can use https://pytorch.org/docs/stable/generated/torch.Tensor.masked_scatter_.html#torch.Tensor.masked_scatter_

?